### DISCUSSION TITLE
HARK AgentType wrapper around Dolo model

### DISCUSSION THREAD

**Discussion by sbenthall** [econ-ark/HARK#1673] [General]
The inverse of trying to export HARK to Dolang (see #659)

This issue is for an AgentType class that:
 - starts with a fully formed Dolo model, like the kind you get from a parsed Dolang YAML file
 - builds all the `getStates`, `getShocks`, etc. methods needed for simulation out of the materials in the Dolo model
 - Ideally, also setting things up so that a solution can be passed in...
 - and can be ran as a simulation.

**Comment by llorracc** [comment]
What exists:

Dolang (yaml) -> Dolo (py-obj)

What this issue is about:

Dolo (py-obj) -> HARK (py-ob)

-----------

Others things that we might do, not in this issue:

HARK -> Dolo

Dolo (py-obj) -> Dolang (yaml)

HARK -> Dolo (py-obj) -> Dolang (yaml) -- might be the best implementation for #446
